### DISCUSSION TITLE
missing unix commands in slim image

### DISCUSSION THREAD

**Discussion by gurolakman** [slimtoolkit/slim#295] [Q&A]
i tried out docker-slim on one of our in-house images & obtained the following results:

REPOSITORY                                                  TAG       IMAGE ID       CREATED          SIZE
com/xyz/sample/docker.slim                  latest    5c2d0aca41c7   3 minutes ago    112MB
com/xyz/sample/docker                       latest    d1394f2120aa   12 minutes ago   724MB

which clearly looked impressive. 

when i tried to make use of the new image (e.g., start a container that uses it) however, i ran into trouble w/ some basic commands used by an in-house <kickstart.sh> bash script not being found.

$ sudo docker run -it -v /tmp:/tmp --name sample -p 8015:8015 --add-host test1:10.0.1.231 --add-host test4:10.0.1.234 com/xyz/sample/docker.slim -native /bin/bash
=== Creating /xyz/admin/sample.conf ===
/xyz/bin/kickstart.sh: line 40: mkdir: command not found
=== Created /xyz/admin/sample.conf:
/xyz/bin/kickstart.sh: line 43: cat: command not found
=== Executing (native) /bin/bash

this makes me think i must be overlooking something rather obvious. any thoughts? 

thanks in advance.

**Comment by eduardosramos** [comment]
I have the same problem:

dokuwiki 02:55:05.56 
dokuwiki 02:55:05.56 Welcome to the Bitnami dokuwiki container
dokuwiki 02:55:05.56 Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-dokuwiki
dokuwiki 02:55:05.57 Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-dokuwiki/issues
dokuwiki 02:55:05.57 
dokuwiki 02:55:05.57 INFO  ==> ** Starting DokuWiki setup **
realpath: /bitnami/apache/conf: No such file or directory
dokuwiki 02:55:05.63 INFO  ==> Configuring Apache ServerTokens directive
dokuwiki 02:55:05.67 INFO  ==> Configuring PHP options
dokuwiki 02:55:05.67 INFO  ==> Setting PHP expose_php option
find: 'false': No such file or directory
dokuwiki 02:55:05.85 INFO  ==> Restoring persisted DokuWiki installation
/opt/bitnami/scripts/libpersistence.sh: line 86: tr: command not found
dokuwiki 02:55:05.85 WARN  ==> No persisted files are configured to be restored
/opt/bitnami/scripts/dokuwiki/entrypoint.sh: line 25: /post-init.sh: No such file or directory

**Comment by mmaker1234** [comment]
The same problem here - even the `ls` and `pwd` commands are missing in an image, based on Ubuntu 20.04

**Reply by kcq** [reply]
@mmaker1234 what kind of application do you have in your image? Also note that the shell and its related components will be removed unless it's used by the application or you ask for it explicitly (e.g., there's a flag called `--include-shell`, which provides a way to keep a basic read-only shell in your minified image).
